libs/libc/stdio/lib_libfwrite.c

```diff
@@ -97,18 +94,11 @@ ssize_t lib_fwrite(FAR const void *ptr, size_t count, FAR FILE *stream)
       goto errout_with_semaphore;
     }
 
-  /* Loop until all of the bytes have been buffered */
+  /* Determine the number of bytes left in the buffer */
 
-  while (count > 0)
+  size_t gulp_size = stream->fs_bufend - stream->fs_bufpos;
```

Comment:
Please define the gulp_size variable at top of function to comply with C89 to get it working for 8-bit MCUs.